### PR TITLE
Remove extraneous MinLen and PolySameLen annotations on Strings in Index JDK

### PR DESCRIPTION
**Pull request by kelloggm**
This is causing the Value Checker to report warnings that it shouldn't when run on Daikon (because #1243 moved the MinLen annotation into the Value Checker). The extraneous PolySameLen annos aren't causing any problems, but they should be removed anyway in case they do cause problems down the line.

### CODE REVIEW

File: checker/jdk/index/src/java/io/File.java [hunk 1]
```diff
@@ -195,7 +195,7 @@ int getPrefixLength() {
      * string for convenience.  This string contains a single character, namely
      * <code>{@link #separatorChar}</code>.
      */
-    public static final @MinLen(1) String separator = "" + separatorChar;
+    public static final /*!MinLen(1)*/ String separator = "" + separatorChar;
```
Comment (smillst) [inline checker/jdk/index/src/java/io/File.java]: Should this just be deleted?

Comment (kelloggm) [inline checker/jdk/index/src/java/io/File.java]: I left it in a comment because it would be correct if the Index Checker could handle Strings. But since MinLen is now a Value Checker annotation, maybe it should just be deleted.

Comment (smillst) [inline checker/jdk/index/src/java/io/File.java]: The problem is that separator has a value type of `@StringVal("/")`, so it can't also have `@MinLen(1)`.

Comment (kelloggm) [inline checker/jdk/index/src/java/io/File.java]: That's going to be a problem when we want to add String support. Since the annotation is commented out, it's not a problem now - but if you'd prefer, I can remove it entirely.

Comment (smillst) [inline checker/jdk/index/src/java/io/File.java]: I don't think any Value Checker annotations should be in the Index jdk.  (Including ones that are commented out.)

Comment (kelloggm) [inline checker/jdk/index/src/java/io/File.java]: We've talked about this before and I agree with you, but that change should be in a separate PR. It's on my TODO list.

Comment (kelloggm) [inline checker/jdk/index/src/java/io/File.java]: I'll remove this one for now though.
